### PR TITLE
Fix storage modification causing bugs

### PR DESCRIPTION
**Pull request by yixiann**
Fixes: https://github.com/AY2223S1-CS2103T-W16-2/tp/issues/297

Removed check of tag duplication in loading data file. If there are duplicate tags in the data file, it is okay. We will create the tags without the duplicates.

If there are items with tags not in the taglist, we will just add them to the taglist instead of saying there is an error

### CODE REVIEW

File: src/test/java/seedu/foodrem/storage/JsonSerializableFoodRemTest.java [hunk 1]
```diff
@@ -52,14 +52,6 @@ public void toModelType_duplicateItems_throwsIllegalValueException() throws Exce
...
     }
 
-    @Test
-    public void toModelType_duplicateTags_throwsIllegalValueException() throws Exception {
-        JsonSerializableFoodRem dataFromFile = JsonUtil.readJsonFile(DUPLICATE_TAG_FILE,
-                JsonSerializableFoodRem.class).get();
-        assertThrows(IllegalArgumentException.class, JsonSerializableFoodRem.MESSAGE_DUPLICATE_TAGS,
-                dataFromFile::toModelType);
-    }
-
```
Comment (yixiann) [inline src/test/java/seedu/foodrem/storage/JsonSerializableFoodRemTest.java:62]: We shall allow duplicate tags in the data file. During conversion these tags will be removed

File: src/test/java/seedu/foodrem/testutil/TypicalFoodRem.java [hunk 2]
```diff
@@ -31,20 +31,12 @@ public static FoodRem getFoodRemWithTypicalItems() {
...
-    /**
-     * Returns a {@code FoodRem} with all the typical tags only.
-     */
-    public static FoodRem getFoodRemWithTypicalTags() {
-        FoodRem foodRem = new FoodRem();
-        for (Tag tag : TypicalTags.getTypicalTags()) {
-            foodRem.addTag(tag);
-        }
-
         return foodRem;
```
Comment (yixiann) [inline src/test/java/seedu/foodrem/testutil/TypicalFoodRem.java:40]: Should have been integrated into the get foodrem typical items

File: src/main/java/seedu/foodrem/model/tag/UniqueTagList.java [hunk 3]
```diff
@@ -110,7 +110,7 @@ public Iterator<Tag> iterator() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof UniqueTagList // instanceof handles nulls
-                && internalList.equals(((UniqueTagList) other).internalList));
+                && new HashSet<>(internalList).equals(new HashSet<>(((UniqueTagList) other).internalList)));
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/tag/UniqueTagList.java:113]: May I know why the reason for this change? Why is there a need for the `HashSet` here?

**Review by bryanljx — DISMISSED** [review]
Looks good!

File: src/main/java/seedu/foodrem/model/tag/UniqueTagList.java (hunk 3 above)
Comment (yixiann) [reply src/main/java/seedu/foodrem/model/tag/UniqueTagList.java:113]: The ordering of the list matters while checking if they are equal. ["vege", "fruits"] will be different from ["fruits", "vege"]

Comment (yixiann) [reply src/main/java/seedu/foodrem/model/tag/UniqueTagList.java:113]: By converting it into a hash set we remove the ordering of the items in the list

Comment (bryanljx) [reply src/main/java/seedu/foodrem/model/tag/UniqueTagList.java:113]: Makes sense, good catch!

**Pull request merged**